python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -533,14 +533,29 @@ def read_from_uri(self, uri: ResourcePath, component: str | None = None, expecte
             # For a local URI allow afw to read it directly.
             return NotImplemented
         pixel_components = ("mask", "image", "variance")
-        if not component or component in pixel_components:
-            # For pixel access download the whole file.
+
+        if component in pixel_components:
+            # For pixel access currently this can not be cached in memory
+            # and the performance gains are unclear. Assume local file
+            # read with file caching for now.
             return NotImplemented
 
-        cached_id, cached_mem = type(self)._cached_fits
-        if self.dataset_ref.id == cached_id:
-            self._reader = self.ReaderClass(cached_mem)
-            return self.readComponent(component)
+        # Cutouts can be optimized. For now only use this optimization
+        # if bbox is the only parameter and the number of pixels in the
+        # bounding box is reasonable.
```

Comment:
This is a good idea, but I don't see the check for number of pixels in the code.

Reply:
I meant to ask you what you thought was reasonable on the PR but then forgot 😄

Reply:
Just off the top of my head, maybe 500x500 or so?  It'd be nice if we knew the compression tile size so we could make it a multiple of that.